autoprotocol/builders.py

```diff
@@ -1738,6 +1747,81 @@ def blowout(self, volume, initial_z, flowrate=None):
             "flowrate": flowrate
         }
 
+    def mode(self, transports=None, mode=None):
+        """Helper for selecting dispense mode based on liquid_class name
+
+        Parameters
+        ----------
+        transports : dict, optional
+            Dictionary of the transport parameters
+        mode : str, optional
+            Mode of dispense type
+
+        Returns
+        -------
+        str
+            Mode of dispense type
+
+        Raises
+        ------
+        ValueError
+            mode does not contain valid mode value
+        TypeError
+            liquid is not of str
+        ValueError
+            liquid_class is not a valid value
+        ValueError
+            multiple liquid_class exists in one LiquidHandle
+        """
+        liquid_classes = set([transport["mode_params"]["liquid_class"]
```

Comment:
Why not define a map of liquid class to dispense type, e.g.
```
liquid_class_to_dispense_modes = {
		"air" : "air_displacement",
		"default" : "positive_displacement",
		"viscous" : "positive_displacement"
	}

liquid_classes = {"air", "default", "viscous"}

dispense_modes = ["air_displacement", "positive_displacement"]

given_liquid_classes = ["air","default","air","viscous","default","air"]

modes = set(map(lambda given_liquid_class: liquid_class_to_dispense_modes[given_liquid_class], given_liquid_classes))

# modes == {'positive_displacement', 'air_displacement'}
```

Reply:
As mentioned earlier, user should have flexibility to choose what dispense mode to use if needed. Liquid handle should be smart enough to resolve based on the liquid class of the sample, but depending on how viscous the sample is (e.g. depending on the concentration of solvent, etc), user should be able to pick whatever `mode` they want.

Reply:
Having that said, I don't mind using dictionary for more legible code...

Reply:
Sure, I think having a default mode mapping/dictionary would definitely make the code more legible. This is a bit tricky to walk through.